### DISCUSSION TITLE
Add ability to add images to a concept

### DISCUSSION THREAD

**Discussion by koenedaele** [OnroerendErfgoed/atramhasis#929] [Ideas]
One of the most requested features has been support for images to illustrate a concept (since 2014). However, the SKOS standard has no support for images. This will make it hard to import thesauri and export them. Although I don't know any thesaurus exporting images, so perhaps it's a purely Atramhasis feature.

A good solution might be linking to images, but not directly uploading them to Atramhasis. This could be very simple such as "provide a URI for an image file", but then we need to determine if we want any metadata on that image to be present as well. A more sensible option might be to allow linking one or more IIIF-manifests (https://iiif.io/) to a concept. This way Atramhasis does not need to become a full-blown DAM, but by including a manifest URI and a client side viewer, people are able to illustrate concepts.

**Comment by koenedaele** [comment]
Something like PAN (Portable Antiquities Netherlands) uses a hasImage property defined at https://data.cultureelerfgoed.nl/vocab/id/pan.html for this. In this case, the property contains the URI of a file, no metadata present.  See https://data.cultureelerfgoed.nl/term/id/pan/15-01-03-95.html for an example of how it's used.

**Comment by goessebr** [comment]
Here are two options suggested by AI. **foaf:depiction** is the most commonly used. **schema:imageObject** (https://schema.org/ImageObject) is more modern and supports additional metadata, though that’s optional.

A custom property like **hasImage** is not recommended because it reduces interoperability.

In the examples below, the image was added within an existing concept, and the image element is marked with the comment `<!-- Image ...` .

## 1. **foaf:depiction**

Best‑practice property for associating an image that *depicts* a resource.  
FOAF does **not** support structured metadata on the image within the property itself, so only the direct URL is shown here.

<details>
<summary>Example (click to expand)</summary>

```xml
<rdf:RDF>
    <rdf:Description rdf:about="https://id.erfgoed.net/thesauri/dateringen/1237">
        <void:inDataset rdf:resource="https://id.erfgoed.net/datasets/thesauri/dateringen"/>
        <dcterms:identifier>1237</dcterms:identifier>
        <skos:inScheme rdf:resource="https://id.erfgoed.net/thesauri/dateringen"/>
        <skos:prefLabel xml:lang="nl-BE">nieuwste tijd</skos:prefLabel>
        <skos:scopeNote xml:lang="nl-BE">
            19th century and later. During this period, Flanders experienced Napoleon,
            a temporary union with the Netherlands, and Belgium's independence in 1830.
            The 20th century is marked by two World Wars, from 1914–1918 and 1939–1945.
        </skos:scopeNote>

        <dcterms:source rdf:nodeID="Na1"/>
        <rdf:type rdf:resource="http://www.w3.org/2004/02/skos/core#Concept"/>
        <skos:narrower rdf:resource="https://id.erfgoed.net/thesauri/dateringen/21"/>
        <skos:narrower rdf:resource="https://id.erfgoed.net/thesauri/dateringen/19"/>
        <skos:narrower rdf:resource="https://id.erfgoed.net/thesauri/dateringen/20"/>
        <skos:related rdf:resource="https://id.erfgoed.net/thesauri/dateringen/1296"/>

        <!-- Image (no metadata possible in FOAF here) -->
        <foaf:depiction rdf:resource="https://images.memorix.nl/rce/thumb/440x440/example.jpg"/>
    </rdf:Description>
</rdf:RDF>
```

</details>

***

## 2. **schema:imageObject**

Use this when you want to store *metadata* with the image, such as license, creator, description, width, height, etc.

<details>
<summary>Example (click to expand)</summary>

```xml
<rdf:RDF>
    <rdf:Description rdf:about="https://id.erfgoed.net/thesauri/dateringen/1237"/>
        <void:inDataset rdf:resource="https://id.erfgoed.net/datasets/thesauri/dateringen"/>
        <dcterms:identifier>1237</dcterms:identifier>
        <skos:inScheme rdf:resource="https://id.erfgoed.net/thesauri/dateringen"/>
        <skos:prefLabel xml:lang="nl-BE">nieuwste tijd</skos:prefLabel>
        <skos:scopeNote xml:lang="nl-BE">
            19th century and later. During this period, Flanders experienced Napoleon,
            a temporary union with the Netherlands, and Belgium's independence in 1830.
            The 20th century is marked by two World Wars, from 1914–1918 and 1939–1945.
        </skos:scopeNote>

        <dcterms:source rdf:nodeID="Na1"/>
        <rdf:type rdf:resource="http://www.w3.org/2004/02/skos/core#Concept"/>
        <skos:narrower rdf:resource="https://id.erfgoed.net/thesauri/dateringen/21"/>
        <skos:narrower rdf:resource="https://id.erfgoed.net/thesauri/dateringen/19"/>
        <skos:narrower rdf:resource="https://id.erfgoed.net/thesauri/dateringen/20"/>
        <skos:related rdf:resource="https://id.erfgoed.net/thesauri/dateringen/1296"/>

        <!-- Image with metadata -->
        <schema:imageObject>
            <schema:contentUrl rdf:resource="https://images.memorix.nl/rce/thumb/440x440/example.jpg"/>
            <schema:license rdf:resource="https://creativecommons.org/licenses/by/4.0/"/>
            <schema:creator>Nationaal Archief</schema:creator>
            <schema:description xml:lang="en">Illustration of the concept “modern period”.</schema:description>
            <schema:width>440</schema:width>
            <schema:height>440</schema:height>
        </schema:imageObject>
    </rdf:Description>
</rdf:RDF>
```

</details>
